src/parser.rs

```diff
@@ -990,18 +994,7 @@ impl<'a> Scanner<'a> {
     }
 
     fn get_segment(&mut self) -> (Segment, bool) {
-        let mut seg = Segment {
-            statements: Vec::new(),
-            next_file: Span::null(),
-            symbols: Vec::new(),
-            local_vars: Vec::new(),
-            global_dvs: Vec::new(),
-            labels: Vec::new(),
-            floats: Vec::new(),
-            buffer: self.buffer_ref.clone(),
-            diagnostics: Vec::new(),
-            span_pool: Vec::new(),
-        };
+        let mut seg = Segment { buffer: self.buffer_ref.clone(), ..Segment::default() };
```

Comment:
Reject, Default doesn't even make sense for this type.  A Segment is externally immutable and has no neutral value.